app/boards/arm/waveshare_rp2040_zero/waveshare_rp2040_zero.dts

```diff
@@ -0,0 +1,162 @@
+/dts-v1/;
+
+#include <rpi_pico/rp2040.dtsi>
+#include "waveshare_rp2040_zero_pinctrl.dtsi"
+#include <freq.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+
+
+/ {
+    model = "Waveshare RP2040 Zero Board";
+    compatible = "waveshare,rp2040-zero", "raspberrypi,rp2040";
+
+
+    chosen {
+        zephyr,code-partition = &code_partition;
+        zephyr,sram = &sram0;
+        zephyr,flash = &flash0;
+    };
+
+    leds {
+        compatible = "gpio-leds";
+        led: led_0 {
+            gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+            label = "LED";
+        };
+    };
+
+    /* These aliases are provided for compatibility with samples */
+    aliases {
+        led0 = &led;
+    };
+
+    xtal_clk: xtal-clk {
+        compatible = "fixed-clock";
+        clock-frequency = <12000000>;
+        #clock-cells = <0>;
+    };
+
+    aliases {
+        watchdog0 = &wdt0;
+    };
+
+
+    zero_header: connector {
+        compatible = "raspberrypi,pico-header";
```

Comment:
This is a compatible that should only be used if this matches the physical layout of a Pi Pico board, which the Zero does not. You'll need a new compatible bindings file for this.